cmd/restic/exclude.go

```diff
@@ -292,3 +293,48 @@ func rejectResticCache(repo *repository.Repository) (RejectByNameFunc, error) {
 		return false
 	}, nil
 }
+
+func rejectBySize(maxSizeStr string) (RejectFunc, error) {
+	maxSize, err := parseSizeStr(maxSizeStr)
+	if err != nil {
+		return nil, err
+	}
+
+	return func(item string, fi os.FileInfo) bool {
+		// directory will be ignored
+		if fi.IsDir() {
+			return false
+		}
+
+		filesize := fi.Size()
+		if filesize > maxSize {
+			debug.Log("file %s is oversize: %d", item, filesize)
+			return true
+		}
+
+		return false
+	}, nil
+}
+
+func parseSizeStr(sizeStr string) (int64, error) {
+	numStr := sizeStr[:len(sizeStr)-1]
+	var unit int64 = 1
+
+	switch sizeStr[len(sizeStr)-1] {
+	case 'b', 'B':
+		// use initialized value, do nothing here
+	case 'k', 'K':
+		unit = 1024
+	case 'm', 'M':
+		unit = 1024 * 1024
+	case 'g', 'G':
+		unit = 1024 * 1024 * 1024
```

Comment:
Please also add handling of `t` and `T` to specify terabytes, both here and in the rest of the code (e.g. tests).